substrate/frame/revive/src/exec.rs

```diff
@@ -1372,13 +1373,88 @@ where
 		}
 		Some(System::<T>::block_hash(&block_number).into())
 	}
-}
 
-/// Determine if the given address is a precompile.
-/// For now, we consider that all addresses between 0x1 and 0xff are reserved for precompiles.
-fn is_precompile(address: &H160) -> bool {
-	let bytes = address.as_bytes();
-	bytes.starts_with(&[0u8; 19]) && bytes[19] != 0
+	fn run_precompile(
+		&mut self,
+		precompile_address: H160,
+		is_delegate: bool,
+		is_read_only: bool,
+		value_transferred: U256,
+		input_data: &[u8],
+	) -> Result<(), ExecError> {
+		if_tracing(|tracer| {
+			tracer.enter_child_span(
+				self.caller().account_id().map(T::AddressMapper::to_address).unwrap_or_default(),
+				precompile_address,
+				is_delegate,
+				is_read_only,
+				value_transferred,
+				&input_data,
+				self.gas_meter().gas_left(),
+			);
+		});
+
+		self.transient_storage.start_transaction();
```

Comment:
We know the pre-compiles can't make use of transient storage. We only pass in the gas meter. No need to have a transaction for that.